File: backend/src/main/java/com/remo/realestatemaintainceoptimizer/entity/Property.java
package com.remo.realestatemaintainceoptimizer.entity;

/**
 * A managed real-estate object ("Objekt"), persisted as one JSON file per property.
 */
public record Property(String id, String name, String address, String icon, Double latitude, Double longitude) {

    /**
     * Creates a property without known coordinates, geocoded on demand via {@code GeocodingService}.
     */
    public Property(String id, String name, String address, String icon) {
        this(id, name, address, icon, null, null);
    }
}

File: frontend/src/types/property.ts
/** A managed real-estate object ("Objekt"), selectable when creating an appointment and listed on the properties page. */
export interface Property {
  id: string
  name: string
  address: string
  icon: string
  latitude: number | null
  longitude: number | null
}

/** Payload for creating a new property. */
export interface CreatePropertyPayload {
  name: string
  address: string
  latitude: number | null
  longitude: number | null
}
